src/deadlock.rs

```diff
@@ -33,10 +37,17 @@
 //! } // only for #[cfg]
 //! ```
 
+use std::marker::PhantomData;
+
 #[cfg(feature = "deadlock_detection")]
 pub use parking_lot_core::deadlock::check_deadlock;
 pub(crate) use parking_lot_core::deadlock::{acquire_resource, release_resource};
 
+#[cfg(not(feature = "deadlock_detection"))]
+pub(crate) struct DeadlockDetectionMarker(PhantomData<()>);
+#[cfg(feature = "deadlock_detection")]
+pub(crate) struct DeadlockDetectionMarker(PhantomData<*const ()>); // !Send
```

Comment:
I made it a struct other than a type alias so that the user can get `DeadlockDetectionMarker` in the !Send error message.

Reply:
Can you check that `MutexGuard<T>` is still `Sync` when `T: Sync`? Add a test for it similar to the `Send` test for the guards.